src/Product.jl

```diff
@@ -0,0 +1,85 @@
+using LinearAlgebra
+using Graphs
+
+struct Product <: AbstractAnsatz
+    tn::Ansatz
+end
+
+Ansatz(tn::Product) = tn.tn
+
+Base.copy(x::Product) = Product(copy(Ansatz(x)))
+Base.similar(x::Product) = Product(similar(Ansatz(x)))
+Base.zero(x::Product) = Product(zero(Ansatz(x)))
+
+function Product(arrays::AbstractArray{<:AbstractVector})
+    n = length(arrays)
+    gen = IndexCounter()
+    symbols = map(arrays) do _
+        nextindex!(gen)
+    end
+    _tensors = map(eachindex(arrays)) do i
+        Tensor(arrays[i], [symbols[i]])
+    end
+
+    sitemap = Dict(Site(i) => symbols[i] for i in eachindex(arrays))
+    qtn = Quantum(TensorNetwork(_tensors), sitemap)
+    graph = Graph(n)
+    mapping = BijectiveIdDict{Site,Int}(Pair{Site,Int}[site => i for (i, site) in enumerate(lanes(qtn))])
+    lattice = Lattice(mapping, graph)
+    ansatz = Ansatz(qtn, lattice)
+    return Product(ansatz)
+end
+
+function Product(arrays::AbstractArray{<:AbstractMatrix})
+    n = length(arrays)
+    gen = IndexCounter()
+    symbols = map(arrays) do _
+        (nextindex!(gen), nextindex!(gen))
+    end
+    _tensors = map(eachindex(arrays)) do i
+        Tensor(arrays[i], [symbols[i][1], symbols[i][2]])
+    end
+
+    sitemap = merge!(
+        Dict(Site(i; dual=true) => symbols[i][1] for i in eachindex(arrays)),
+        Dict(Site(i) => symbols[i][2] for i in eachindex(arrays)),
+    )
+    qtn = Quantum(TensorNetwork(_tensors), sitemap)
+    graph = Graph(n)
+    mapping = BijectiveIdDict{Site,Int}(Pair{Site,Int}[site => i for (i, site) in enumerate(lanes(qtn))])
+    lattice = Lattice(mapping, graph)
+    ansatz = Ansatz(qtn, lattice)
+    return Product(ansatz)
+end
+
+LinearAlgebra.norm(tn::Product, p::Real=2) = LinearAlgebra.norm(socket(tn), tn, p)
```

Comment:
```suggestion
"""
    LinearAlgebra.norm(tn::Product, p::Real=2)

Compute the p-norm of a [`Product`](@ref) ansatz.
"""
LinearAlgebra.norm(tn::Product, p::Real=2) = LinearAlgebra.norm(socket(tn), tn, p)
```

Reply:
as with other comments, we don't need a docstring here because the semantics are the same